symengine/tests/test_var.py

```diff
@@ -38,6 +38,13 @@ def test_var():
     # check return value
     assert v == [d, e, fg]
 
+# This works with nosetests,
+# but not with py.test, unclear why.
+# nosetests ignores pytest.mark.xfail so
+# we can safely decorate this function with this
+import pytest
+@pytest.mark.xfail
+def test_var_global_namespace():
     # see if var() really injects into global namespace
     raises(NameError, lambda: z1)
     _make_z1()
```

Comment:
I just don't like hackish solutions, and this feel very hackish, with unclear solution in sight. If you want to switch to py.test later, that's fine too, but then we should simply remove any py.test from this PR.

The reason I am saying that is that if you become busy after this PR is in, then we'll end up with a half baked transition between nosetests and py.test that somebody else will have to spend time to understand and untangle.

I am fine either way to use or not use py.test, but let's stick to one and follow through. I don't want to use both with different workarounds for each.